sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -143,6 +149,117 @@ internal class ViewAnnotationManagerImpl(
     return renderThread.viewAnnotationMode
   }
 
+  override fun showAnnotations(
+    annotations: List<View>,
+    edgeInsets: EdgeInsets?,
+    bearing: Double?,
+    pitch: Double?,
+    animationDurationMs: Long
+  ) {
+    if (annotations.isNotEmpty()) {
+      val viewAnnotationOptions = annotations.mapNotNull {
+        getViewAnnotationOptionsByView(it)
+      }
+      val coordinateBounds = coordinateBoundsFromAnnotations(viewAnnotationOptions)
+      val paddings = calculateEdgeInset(viewAnnotationOptions, edgeInsets)
+      val geometry = GeometryCollection.fromGeometries(
+        listOf(
+          coordinateBounds.northeast,
+          coordinateBounds.southwest
+        )
+      )
+      // create overview viewport state with geometry, padding and animation duration
+      val overviewViewportState = mapView.viewport.makeOverviewViewportState(
+        OverviewViewportStateOptions.Builder()
+          .geometry(geometry)
+          .padding(paddings)
+          .animationDurationMs(animationDurationMs)
+          .build()
+      )
+      // transition the map camera to target overview viewport state.
+      mapView.viewport.transitionTo(
+        targetState = overviewViewportState,
+        completionListener = { success ->
+          if (!success) {
+            logI(TAG, "transition to annotations failed")
+          }
+        }
+      )
+    } else {
+      logW(TAG, "Empty annotation list provided.")
+    }
+  }
+
+  override fun cameraForAnnotations(
+    annotations: List<View>,
+    edgeInsets: EdgeInsets?,
+    bearing: Double?,
+    pitch: Double?
+  ): CameraOptions? {
+    if (annotations.isNotEmpty()) {
+      val viewAnnotationOptions = annotations.mapNotNull {
+        getViewAnnotationOptionsByView(it)
+      }
+      val coordinateBounds = coordinateBoundsFromAnnotations(viewAnnotationOptions)
+      val paddings = calculateEdgeInset(viewAnnotationOptions, edgeInsets)
+      return mapboxMap.cameraForCoordinateBounds(
+        coordinateBounds,
+        paddings,
+        bearing,
+        pitch
+      )
+    }
+    return null
+  }
+
+  /**
+   * Convenience function to get bounds occupied by list of [ViewAnnotationOptions]'s geometry.
+   */
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  internal fun coordinateBoundsFromAnnotations(annotationOptions: List<ViewAnnotationOptions>): CoordinateBounds {
+    var north = -MAX_LATITUDE
+    var south = MAX_LATITUDE
+    var west = MAX_LONGITUDE
+    var east = -MAX_LONGITUDE
+    annotationOptions.map {
+      it.geometry?.let { geometry ->
+        val point = Point.fromJson(geometry.toJson())
+        north = maxOf(north, point.latitude())
+        south = minOf(south, point.latitude())
+        west = minOf(west, point.longitude())
+        east = maxOf(east, point.longitude())
```

Comment:
Does this algorithm work if I have two annotations, one with longitude = -170 and the other one with longitude 170?

I guess it would return a very large longitude span (340º) when instead I would expect it to be 20º.

What do we usually do for this cases?

Reply:
Yeah, correct. it does return large longitude. Normally, in case of camera movements we calculate the shortest path between points. Need to check if we do compensate when crossing antimeridian for camera bounds.

Reply:
Looked a bit into that and i think instead of calculating camerabounds here, we could just use `map.cameraForCoordinates` and let gl-native handle bounds (although that method also gives longitudnal span of 340 for this case, and doesn't consider through antimeridian, but will confirm this with native too). I will update pr accordingly :)